src/mpp/Dec.hpp

```diff
@@ -239,7 +259,7 @@ struct Resolver {
 	static constexpr size_t SIZE = path_item_static_size(PI);
 	static constexpr size_t USR_ARG_COUNT = path_item_static_size(P0);
 	static_assert(path_item_type(P0) == PIT_STATIC_L0);
-	static_assert((is_path_item_static(PI) && POS < SIZE) ||
+	static_assert(((is_path_item_static(PI) || path_item_type(PI) == PIT_VARIANT) && POS < SIZE) ||
 		      (!is_path_item_static(PI) && POS + SIZE == 0));
```

Comment:
This assert becomes overly complicated, and it's hard to understand. Could you please leave a comment or add a helper?

Reply:
Clarified the assertion.